src/@types/DragThoughtItem.ts

```diff
@@ -7,6 +7,7 @@ interface DragThoughtItem {
   path: Path
   simplePath: SimplePath
   zone: DragThoughtZone
+  type: string
```

Comment:
I had to put it because `useDrag` requires the drag item to atleast have a type property, and since we were using `DragThoughtItem` I had to put the `type` there.

Reply:
Got it. I reread the react-dnd [documentation](https://react-dnd.github.io/react-dnd/docs/overview) on items and types, and it makes sense now.

`string` is a **broad** type that includes all strings.
`'thought' | 'toolbar-button'` is a **narrow** type that only includes the literal values `'thought'` and `'toolbar-button'`.

In general, types should be as narrow as possible, while still encompassing all the valid values.

As for inheritance, a better approach would be to extend the existing react-dnd `DragObjectWithType`:

```ts
interface DragThoughtItem extends DragObjectWithType {
  path: Path
  simplePath: SimplePath
  zone: DragThoughtZone
}
```

Reply:
Hmm, I guess its better to create an enum, inheriting from `DragObjectWithType` would lead to using type `string | symbol`.

Reply:
Yes, good point. 

It should just be `'thought' | 'toolbar-button'`. Favorites are thoughts, and eventually we should be able to drag favorites to/from the editor.